### PR TITLE
Fixed #25853 -- Added support for GeoHash function on SpatiaLite.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/25853

### CODE REVIEW

File: django/contrib/gis/db/backends/spatialite/operations.py [hunk 1]
```diff
@@ -216,6 +219,10 @@ def spatialite_version(self):
         "Returns the SpatiaLite library version as a string."
         return self._get_spatialite_func('spatialite_version()')
 
+    def lwgeom_version(self):
+        """Return the version of LWGEOM library used by SpatiaLite."""
+        return self._get_spatialite_func('spatialite_version()')
```
Comment (charettes) [inline django/contrib/gis/db/backends/spatialite/operations.py]: Wrong func?

Comment (sir-sigurd) [inline django/contrib/gis/db/backends/spatialite/operations.py]: Yes, many thanks.

File: docs/releases/1.10.txt [hunk 2]
```diff
@@ -82,6 +82,10 @@ Minor features
 * Added support for the :class:`~django.contrib.gis.db.models.MakeLine`
   aggregate on SpatiaLite.
 
+* Added support for the
+  :class:`~django.contrib.gis.db.models.functions.GeoHash`
+  function on SpatiaLite.
+
```
Comment (claudep) [inline docs/releases/1.10.txt]: I think we could merge this item with the one above.

**Comment by timgraham** [comment]
Looks like the failing test is because `self.lwgeom_version()` returns `None` even though SpatiaLite is 4.1.1. Do we need to install an additional package or something?

**Comment by sir-sigurd** [comment]
@timgraham 
The test failed because I made a faulty check of `lwgeom_version()`, it's fixed now. It looks like installing liblwgeom is not sufficient and SpatiaLite should be compiled with LWGEOM to support GeoHash function as it done in the latest Ubuntu releases. I don't know if we should merge this until we have SpatiaLite compiled with LWGEOM on Jenkins.

**Comment by claudep** [comment]
I think it's fine if the test pass with or without LWGEOM compiled in.

**Comment by sir-sigurd** [comment]
@claudep 
I tested it locally both with and without LWGEOM support and test passed.

**Comment by timgraham** [comment]
merged in 25f5b5c19dc23c6c09877c9c652a934e9669573d, thanks!

**Pull request closed without merging**